metricbeat/module/docker/docker.go

```diff
@@ -133,18 +133,38 @@ func FetchStats(client *client.Client, timeout time.Duration) ([]Stat, error) {
 // This is currently very inefficient as docker calculates the average for each request,
 // means each request will take at least 2s: https://github.com/docker/docker/blob/master/cli/command/container/stats_helpers.go#L148
 // Getting all stats at once is implemented here: https://github.com/docker/docker/pull/25361
-func exportContainerStats(ctx context.Context, client *client.Client, container *types.Container) Stat {
+// In case stream is true, we use get a stream of results for container stats. From the stream we keep the second result.
+// This is needed for podman use case where in case stream is false, no precpu stats are returned. The precpu stats
+// are required for the cpu percentage calculation. We keep the second  result as in the first result, the stats are not correct.
+func exportContainerStats(ctx context.Context, client *client.Client, container *types.Container, stream bool) Stat {
 	var event Stat
 	event.Container = container
-
-	containerStats, err := client.ContainerStats(ctx, container.ID, false)
+	containerStats, err := client.ContainerStats(ctx, container.ID, stream)
 	if err != nil {
 		return event
 	}
-
 	defer containerStats.Body.Close()
-	decoder := json.NewDecoder(containerStats.Body)
-	decoder.Decode(&event.Stats)
 
+	// JSON decoder
+	decoder := json.NewDecoder(containerStats.Body)
+	if !stream {
+		if err := decoder.Decode(&event.Stats); err != nil {
+			return event
+		}
+	} else {
```

Comment:
Looks like in either case (error or not) we return the content of the `event` variable here. Does it even make sense to check for the error in this case?

Reply:
You are right, I was just receiving a linter error. I decided to add an debug log message.